### ISSUE TITLE
bug: (YTM) some songs get scrobbled twice, no discernable pattern

### ISSUE THREAD

**Issue by xathon** [FoxxMD/multi-scrobbler#228]
### Please check existing knowledge before opening an issue

- [X] I have [checked the FAQ](https://foxxmd.github.io/multi-scrobbler/docs/FAQ/) and [documentation](https://foxxmd.github.io/multi-scrobbler/docs/configuration/)
- [X] I have [searched through existing issues](https://github.com/FoxxMD/multi-scrobbler/issues?q=sort%3Aupdated-desc) and [discussions.](https://github.com/FoxxMD/multi-scrobbler/discussions)

### Describe the Bug

Some songs from YouTube Music get scrobbled twice. Sometimes Maloja detects this and rejects the scrobble (like with the song Feuertaufe by In Extremo in this log excerpt), but most of the time it just shows up duplicated.

I've attached a log (mentions of Spotify removed due to relevance) where I restarted the container, manually reauthenticated, and just played around a bit and skipped songs in the middle, in the beginning, paused for a bit, etc., but never went backwards. The only thing close to a pattern I've been seeing is that these duplicates seem to show up more often if I have paused playback for a bit and then skipped to the next song, or if it's the first song in a while, but this isn't 100% either.

Let me know if there are any other things I should test.

Maloja Scrobbles page:
![image](https://github.com/user-attachments/assets/d4793fae-24ce-45ab-a497-c6a1d315b98e)

YTM History (ignore top 2, I kept listening; but note how far down A Rash Decision is vs Maloja/Scrobbler Logs):
![image](https://github.com/user-attachments/assets/b835c414-6ecb-4a16-8427-6216a6dcb645)

### Platform

Docker

### Versions

- multi-scrobbler: docker image foxxmd/multi-scrobbler:pr-203 (for yt auth)
- maloja: 3.2.2

### Logs

```shell
too long, see https://termbin.com/g8si 
```

### Additional Context

The log also shows instances of #226 and #227.

**Comment by xathon** [comment]
[logs.txt](https://github.com/user-attachments/files/17709585/logs.txt)
Alternatively, logs as download here

**Comment by FoxxMD** [comment]
Thanks for the report and detailed logs. Just want to acknowledge I've seen this. End of the year is a busy time for me (socially). The YTM auth issues have been a larger priority but I will get to this soon :tm:

**Comment by xathon** [comment]
Oh no worries! I did throw a lot at you :D thanks for the ack, hope you have some quality time!

**Comment by xathon** [comment]
Linking my comment from the other issue here as the same thing happens and the logs are a bit more detailed:

https://github.com/FoxxMD/multi-scrobbler/issues/227#issuecomment-2533208376

**Comment by nrosier** [comment]
Also noticed this issue. I just manually remove the duplicates until this gets fixed.

**Comment by FoxxMD** [comment]
@xathon @nrosier please try docker image `foxxmd/multi-scrobbler:pr-248` with `logDiff: true` and let me know if this fixes anything for you. The changes in PR #248 were made mainly with #227 in mind but they may also address this and #226

**Comment by nrosier** [comment]
@FoxxMD been using pr248-dd4b2eb for a couple of days now. about 50 scrobbles and no duplicates so far. usually I had some every 10-20 scrobbles so it's looking promising.

**Comment by FoxxMD** [comment]
Hi @xathon @nrosier any more updates on whether this fix is holding?

**Comment by xathon** [comment]
Looking good so far, the regular duplicate scrobbles are gone! I did discover what's probably an edge case - I think I listened to a song at like 00:06 today, but didn't finish it. When I opened YTM just now, it was the song that was in Now Playing, but at 0:00. It did show up twice in the scrobbeled tracks though, but both as 22 hours ago. I'll get the logs and attack them here.

**Comment by FoxxMD** [comment]
I don't think that is avoidable, unfortunately. Since YTM does not timestamp anything and MS is at the mercy of this one list...if YTM decides to bump that track because its its in now playing when you opened the browser then MS is going to see that as a play.

**Comment by FoxxMD** [comment]
Fixed in [v0.9.0](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.9.0)

**Comment by xathon** [comment]
Yeah, agree that this probably can't be avoided. Just wanted to mention it since it fit the issue at hand.

But yeah, I think the other issues were related to that as well. Thanks a lot for your work!

**Comment by nrosier** [comment]
I have over 150 scrobbles with 0 duplicates so this seems to fix it. Thanks a lot for the great work.
